src/status_im/chat/events.cljs

```diff
@@ -224,31 +224,33 @@
 ;;TODO (yenda) find a more elegant solution for system messages
 (defn- mark-messages-seen
   [chat-id {:keys [db] :as cofx}]
-  (let [me                         (:current-public-key db)
-        messages-path              [:chats chat-id :messages]
-        unseen-messages-ids        (into #{}
-                                         (comp (filter (fn [[_ {:keys [from user-statuses outgoing]}]]
-                                                         (and (not outgoing)
-                                                              (not= constants/system from)
-                                                              (not= (get user-statuses me) :seen))))
-                                               (map first))
-                                         (get-in db messages-path))
-        unseen-system-messages-ids (into #{}
-                                         (comp (filter (fn [[_ {:keys [from user-statuses]}]]
-                                                         (and (= constants/system from)
-                                                              (not= (get user-statuses me) :seen))))
-                                               (map first))
-                                         (get-in db messages-path))]
-    (when (or (seq unseen-messages-ids)
-              (seq unseen-system-messages-ids))
-      (handlers-macro/merge-fx cofx
-                               {:db (-> (reduce (fn [new-db message-id]
-                                                  (assoc-in new-db (into messages-path [message-id :user-statuses me]) :seen))
-                                                db
-                                                (into unseen-messages-ids unseen-system-messages-ids))
-                                        (update-in [:chats chat-id :unviewed-messages] set/difference unseen-messages-ids unseen-system-messages-ids))}
-                               (persist-seen-messages chat-id (into unseen-messages-ids unseen-system-messages-ids))
-                               (send-messages-seen chat-id unseen-messages-ids)))))
+  (if (get-in db [:chats chat-id :public?])
```

Comment:
Could you add a `public-chat?` method here https://github.com/status-im/status-react/blob/beb7cd2b84efa03d893a42e392c5550dde38369b/src/status_im/chat/models.cljs#L15 ? In the same way as https://github.com/status-im/status-react/blob/79aadec53edcf2907586c90ab0cd356c0352755e/src/status_im/chat/models.cljs#L19 .

Reply:
yes sure